File: cpp/04-Queue/10845.cpp
#include <iostream>
#include <queue>
#include <string>

using namespace std;

int main()
{
    ios::sync_with_stdio(0);
    cin.tie(0);

    int N;
    cin >> N;

    queue<int> Q;
    for(int i = N; i > 0; i--)
    {
        string command;
        cin >> command;

        if(command == "push")
        {
            int value;
            cin >> value;
            Q.push(value);
        }

        else if(command == "pop")
        {
            if(Q.empty())
                cout << -1 << "\n";
            else
            {
                cout << Q.front() << "\n";
                Q.pop();
            }
        }

        else if(command == "size")
        {
            cout << Q.size() << "\n";
        }


        else if(command == "empty")
        {
            if(Q.empty())
                cout << 1 << "\n";
            else
                cout << 0 << "\n";
        }

        else if(command == "front")
        {
            if(Q.empty())
                cout << -1 << '\n';
            else
                cout << Q.front() << '\n';
        }

        else if(command == "back")
        {
            if(Q.empty())
                cout << -1 << '\n';
            else
                cout << Q.back() << '\n';
        }

    }

    return 0;
}

File: python/0x06_Queue/10845.py
import sys
from collections import deque

def solution():
    queue = deque()
    
    N = int(sys.stdin.readline())
    commands = [sys.stdin.readline().rstrip() for _ in range(N)]
    
    for command in commands:
        command = command.split(' ')

        if command[0] == "push":
            queue.append(command[1])
        
        elif command[0] == 'pop':
            if queue:
                v = queue.popleft()
                print(v)
            else:
                print(-1)
        
        elif command[0] == 'size':
            print(len(queue))

        elif command[0] == 'empty':
            if not queue:
                print(1)
            else:
                print(0)
        
        elif command[0] == 'front':
            if queue:
                print(queue[0])
            else:
                print(-1)
        
        elif command[0] == 'back':
            if queue:
                print(queue[-1])
            else:
                print(-1)

if __name__ == "__main__":
    solution()